src/Item.cpp

```diff
@@ -3,27 +3,47 @@
 
 namespace tt
 {
-
-    Item::Item( sf::Texture texture,
-                const sf::Vector2i& size,
-                nl::json json)
+    Item::Item(	const std::string&  id,
+               	sf::Texture         texture,
+               	const sf::Vector2i& size,
+               	const nl::json&     json )
         : AnimatedSprite(texture, size),
-            _json { json }
+            _json   { json },
+            _id     { id }
     {
 
     }
 
-    std::string Item::getDescription() { 
+    std::string Item::getID() const
+    {
+        return _id;
+    }
+
+    std::string Item::getName() const
+    {
 
         //
         // Why is it that the json object can be treated as a top level
         // associative array when this is created in ItemFactory but 
         // after it is assigned to the member variable in this object
         // instance, it becomes a json array at the top level, 
-        // and we must use _json[0] to get the
-        // first item?
+        // and we must use _json[0] to get the first item?
         //
-        return (std::string)(_json[0].at("description"));
+        return _json[0].at("name");
+    }
+
+    std::string Item::getDescription() const
+    {
+        return _json[0].at("description");
```

Comment:
I'm ok with the way this currently is, but in the future it would be better to save the values in member variables and return those. 

Doing so would save the two index lookups that happen in this call. Granted, they are both O(1) lookups, but zero lookups is still better than two.